JP-3948: Use wavelengthrange reference file information in white light step

emolter opened the pull request:
<!-- If this PR closes a JIRA ticket, make sure the title starts with the JIRA issue number,
for example JP-1234: <Fix a bug> -->
Resolves [JP-3948](https://jira.stsci.edu/browse/JP-3948)

<!-- If this PR closes a GitHub issue that is not attached to a Jira ticket, uncomment the line below, and reference it here by its number -->
Closes #9311 

<!-- describe the changes comprising this PR here -->
This PR allows the white light step to read in different default wavelength ranges for different orders/filters from a reference file for NIRISS SOSS data.

<!-- if you can't perform these tasks due to permissions, please ask a maintainer to do them -->
## Tasks
- [ ] If you have a specific reviewer in mind, tag them.
- [x] add a build milestone, i.e. `Build 12.0` (use the [latest build](https://github.com/spacetelescope/jwst/milestones) if not sure)
- [x] Does this PR change user-facing code / API? (if not, label with `no-changelog-entry-needed`)
  - [x] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see [changelog readme](https://github.com/spacetelescope/jwst/blob/main/changes/README.rst) for instructions) 
  - [x] update or add relevant tests
  - [x] update relevant docstrings and / or `docs/` page
  - [x] [start a regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/jwst.yml) and include a link to the running job ([click here for instructions](https://github.com/spacetelescope/RegressionTests/blob/main/docs/running_regression_tests.md))
    - [x] Do truth files need to be updated ("okified")?
      - [ ] **after the reviewer has approved these changes**, run `okify_regtests` to update the truth files
- [ ] if a JIRA ticket exists, [make sure it is resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

emolter commented:
regtests https://github.com/spacetelescope/RegressionTests/actions/runs/16302336812

emolter commented on docs/jwst/white_light/reference_files.rst:
```diff
@@ -1,5 +1,8 @@
 Reference File
 ==============
-The ``white_light`` step uses the PARS-WHITELIGHTSTEP parameter reference file.
```
Am I correct in saying we don't typically document the parameter reference files for every step, and this can be removed?  If that's misguided I will put it back.

melanieclarke replied:
I agree, we don't typically document the parameter reference files.

emolter commented on jwst/white_light/white_light.py:
```diff
@@ -174,3 +181,55 @@ def _make_empty_output_table(input_model):
+
+    If user-specified min/max wavelengths are provided, they will be used.
+    Otherwise, the function will look for the wavelength range in the reference file info.
+    If no reference file is provided, the entire wavelength range will be used.
```
This feels like a distinction without a difference.  The fallback values are broad enough that the entire wavelength range _will_ indeed be used.  And I'm not convinced something like

> If no reference file range is found, broad fallback values are returned, such that all wavelengths found in the input model spectral tables are used

is clearer than the current description

emolter commented:
Hopefully https://github.com/spacetelescope/jwst/pull/9650/commits/541e92b72e72b3ce6479fdfce60b37fac20726e4 addresses all the comments, let me know if I missed anything
